gui/zoom.py

```diff
@@ -0,0 +1,58 @@
+"""
+Zoom
+====
+Module provides zoom functionality for a tkinter canvas.
+
+This module includes functions to start panning, perform panning, and zoom in/out on a tkinter canvas.
+It also includes a function to bind these events to the canvas.
+
+Functions:
+        start_pan(event): Marks the starting point for panning.
+        do_pan(event): Drags the canvas to the new position.
+        zoom(event): Zooms in or out on the canvas.
+        bind_canvas_zoom_events(canvas): Binds the pan and zoom events to the canvas.
+
+Example:
+    import tkinter as tk
+    from gui import bind_canvas_zoom_events
+
+    # Create a tkinter window and canvas
+    window = tk.Tk()
+    canvas = tk.Canvas(window, bg="white", width=600, height=450)
+    canvas.pack() # or grid()
+
+    # Set the canvas and bind events
+    bind_canvas_zoom_events(canvas)
+
+    # Run the tkinter event loop
+    window.mainloop()
+"""
+import tkinter as tk
+
+
+def start_pan(event: tk.Event):
+    """Marks the starting point for panning."""
+    canvas: tk.Canvas = event.widget
+    canvas.scan_mark(event.x, event.y)
+
+
+def do_pan(event: tk.Event):
+    """Drags the canvas to the new position."""
+    canvas: tk.Canvas = event.widget
+    canvas.scan_dragto(event.x, event.y, gain=1)
+
+
+def zoom(event: tk.Event):
+    """Zooms in or out on the canvas."""
```

Comment:
With the adjustment for pan

Reply:
Thanks for pointing it out. This was left out as original. I'll update the docstring with the upcoming commit.

Reply:
@Iv-Tatiana Can you check out whether the documentation is clear enough now (and correct)?

Reply:
Yes! Thank you!